### DISCUSSION TITLE
VBoxLinuxAdditions for Orchid

### DISCUSSION THREAD

**Discussion by Thatoo** [Vanilla-OS/handbook#101] [Q&A]
I tried to follow https://handbook.vanillaos.org/2023/01/21/install-in-vm.html#title15 but it didn't work.
First `sudo abroot sheel` doesn't do anything.
I wanted to make a PR because it needs to change
`cp -r /media/<username>/Vbox_GAs_<version> /tmp/vbox`
into
`cp -r /media/<username>/VBox_GAs_<version> /tmp/vbox`
but I'm afraid this tutorial isn't for Orchid and it might need to be rewritten I guess.

`sudo abroot shell` doesn't work.
`./VBoxLinuxAdditions.run` failes and
` /sbin/rcvboxadd quicksetup all` followed by reboot don't make VBoxLinuxAdditions works in Orchid.

**Comment by kbdharun** [comment]
> I tried to follow https://handbook.vanillaos.org/2023/01/21/install-in-vm.html#title15 but it didn't work. First `sudo abroot sheel` doesn't do anything. I wanted to make a PR because it needs to change `cp -r /media/<username>/Vbox_GAs_<version> /tmp/vbox` into `cp -r /media/<username>/VBox_GAs_<version> /tmp/vbox` but I'm afraid this tutorial isn't for Orchid and it might need to be rewritten I guess.

Yes, the tutorial as mentioned in the guide's header hasn't been updated for Orchid. ABRoot shell functionality to make changes to the root partition directly has been removed, it is suggested to use a [custom image](https://github.com/Vanilla-OS/custom-image) for your needs instead.

> `sudo abroot shell` doesn't work. `./VBoxLinuxAdditions.run` failes and ` /sbin/rcvboxadd quicksetup all` followed by reboot don't make VBoxLinuxAdditions works in Orchid.

Same as above, we ship Debian's VBox guest additions package already as part of the VM image which you can choose during installation or by updating `host-shell pkexec nano /etc/abroot/abroot.json` files image value from desktop to vm after which you can perform an upgrade using `host-shell pkexec abroot upgrade`.

**Comment by Thatoo** [comment]
I did install the VM version.
Is that correct?
![image](https://github.com/Vanilla-OS/handbook/assets/464782/fe5efe7c-3ce8-4a36-946e-4cf5297b1a3f)

However, I can't copy paste from host to invited, I can't full screen the VM. So it doesn't behave as if VBox guest additions was installed in Orchid # 113 or I missed a step somewhere.

**Reply by kbdharun** [reply]
> However, I can't copy paste from host to invited, I can't full screen the VM. So it doesn't behave as if VBox guest additions was installed in Orchid # 113 or I missed a step somewhere.

I haven't tested moving from desktop image to VM image previously. It could be the kernel module for guest additions isn't getting properly loaded, maybe try `host-shell pkexec abroot upgrade --force`. Or else I would suggest creating a new VM where you can select the VM image from the installer and then you will have clipboard sharing and other features.

**Reply by Thatoo** [reply]
The thing is that I install the VM version at install (ISO beta # 113).
I tried a `host-shell pkexec abroot upgrade --force` but I still can't have clipboard sharing and other features.
I'll try installing a new VM when a new ISO (more recent than # 117) will be released because # 117 can't be installed in VM as  explain in https://github.com/Vanilla-OS/Albius/issues/47 , https://github.com/Vanilla-OS/Albius/issues/48 and https://github.com/Vanilla-OS/Albius/issues/49 .
